### ISSUE TITLE
git@github.com: Permission denied (publickey)

### ISSUE THREAD

**Issue by vinayakg** [Homebrew/brew#7576]
- [x] ran `brew update` and can still reproduce the problem?
- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [x] ran `brew config` and `brew doctor` and included their output with your issue?

## What you were trying to do (and why)

I had cleaned up my git creds and setup all using ssh access (which I blogged [here](https://vinayakg.dev/staying-sane-using-multiple-git-accounts)) and after a couple of days I ran into this issue when I ran `brew update`

> git@github.com: Permission denied (publickey)

I was trying to see if there is way to set the git remote for the repository, but I did not find any help.

## What happened (include command output)
Got error with `brew update`

> git@github.com: Permission denied (publickey)

<details>
  <summary>Command output</summary>
  <pre>
 ✔  brew update
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
Error: Fetching /usr/local/Homebrew/Library/Taps/delaguardo/homebrew-graalvm failed!

  </pre>
</details>

## What you expected to happen

Expected it to work and fetch the newest version of Homebrew and all formulae from GitHub using git

## Step-by-step reproduction instructions (by running `brew` commands)

Run the `brew update` command

## Output of `brew config` and `brew doctor` commands

<pre>

 ✔  brew update
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
Error: Fetching /usr/local/Homebrew/Library/Taps/delaguardo/homebrew-graalvm failed!

 ✔  brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: A newer Command Line Tools release is available.
Update them from Software Update in System Preferences or run:
  softwareupdate --all --install --force

If that doesn't show you an update run:
  sudo rm -rf /Library/Developer/CommandLineTools
  sudo xcode-select --install

Alternatively, manually download them from:
  https://developer.apple.com/download/more/.

Warning: You have unlinked kegs in your Cellar.
Leaving kegs unlinked can lead to build-trouble and cause brews that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  vim
  shared-mime-info
  pdftk

Warning: Your Xcode (11.3.1) is outdated.
Please update to Xcode 11.4.1 (or delete it).
Xcode can be updated from the App Store.

</pre>

**Comment by vinayakg** [comment]
Then I was landed #52, which gave me some clue to look at taps, so I ran `brew tap-info --installed` to see the git urls involved from the taps

<pre>
 ↵ 128  brew tap-info --installed
adoptopenjdk/openjdk: unpinned, 32 casks
/usr/local/Homebrew/Library/Taps/adoptopenjdk/homebrew-openjdk (418 files, 326.0KB)
From: https://github.com/adoptopenjdk/homebrew-openjdk

delaguardo/graalvm: unpinned, private, 3 casks
/usr/local/Homebrew/Library/Taps/delaguardo/homebrew-graalvm (31 files, 38.3KB)
From: git@github.com:DeLaGuardo/homebrew-graalvm.git

homebrew/cask: unpinned, 1 command, 3541 casks
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask (4,015 files, 234.1MB)
From: https://github.com/Homebrew/homebrew-cask

homebrew/cask-fonts: unpinned, 924 casks
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask-fonts (1,322 files, 3.6MB)
From: https://github.com/Homebrew/homebrew-cask-fonts

homebrew/cask-versions: unpinned, 152 casks
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask-versions (2,285 files, 58.6MB)
  1 Host *
From: https://github.com/Homebrew/homebrew-cask-versions

homebrew/core: unpinned, 2 commands, 4997 formulae
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core (5,558 files, 35.6MB)
From: https://github.com/Homebrew/homebrew-core

homebrew/services: unpinned, 1 command
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-services (127 files, 125.5KB)
From: https://github.com/Homebrew/homebrew-services

rs/tap: unpinned, 4 formulae
/usr/local/Homebrew/Library/Taps/rs/homebrew-tap (33 files, 45.7KB)
From: https://github.com/rs/homebrew-tap

sambadevi/powerlevel9k: unpinned, 11 formulae
/usr/local/Homebrew/Library/Taps/sambadevi/homebrew-powerlevel9k (52 files, 43.8KB)
From: https://github.com/sambadevi/homebrew-powerlevel9k

spl/pdftk: unpinned, 1 formula
/usr/local/Homebrew/Library/Taps/spl/homebrew-pdftk (30 files, 46.7KB)
From: https://github.com/spl/homebrew-pdftk
</pre>

This gave some clues on the problem, understood that I dont have alias github.com, I use gh and gh-work as seen below

```
Host gh
 HostName github.com
 User git
 UseKeychain yes
 IdentityFile ~/.ssh/id_vinayakg_rsa

Host gh-pyt
  HostName github.com
  User git
  UseKeychain yes
  IdentityFile ~/.ssh/id_pyt_rsa
```
The immediate solution was to add one more config duplicating one of the above config with github.com and I expected it work and it did work.

I am trying to understand if there is an easier way to edit the git remote url for the taps installed on my machine. Let me know if that is possible or I will close the issue considering the ssh solution works.

**Comment by MikeMcQuaid** [comment]
> [git@github.com](mailto:git@github.com): Permission denied (publickey)

We never clone using SSH (`git@`) so this isn't a Homebrew issue and your connection configuration to GitHub is up to you.
